pkg/apis/middleware/session.go

```diff
@@ -17,7 +17,7 @@ type VerifyFunc func(ctx context.Context, token string) (*oidc.IDToken, error)
 
 // CreateTokenToSessionFunc provides a handler that is a default implementation
 // for converting a JWT into a session.
-func CreateTokenToSessionFunc(verify VerifyFunc) TokenToSessionFunc {
+func CreateTokenToSessionFunc(verify VerifyFunc, allowUnverifiedEmail bool) TokenToSessionFunc {
```

Comment:
Overall the code quality and the tests you included look good but I think logically this seems to be the wrong location for this feature. We should do some refactoring on this.

Reply:
Instead of adding a new parameter to the CreateTokenToSessionFunc we should extend the VerifyFunc as it is the correct places for new verification logic.

We will need to extend the `IDTokenVerificationOptions` struct with a new field `AllowUnverifiedEmail bool` in the `pkg > providers > oidc > verifier.go`.

Furthermore, the `toVerificationOptions()` in `pkg > providers > oidc > provider_verifier.go` needs to be extended with the new field.